akka-stream/src/main/scala/akka/stream/scaladsl/One2OneBidiFlow.scala

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream._
+
+import scala.util.control.NoStackTrace
+
+/**
+ * A generic BidiFlow which verifies that another flow produces exactly one output element per
```

Comment:
Excellent!
Thanks, @drewhk!
I'll rewrite the whole thing based on the `GraphStage`, which should nicely allow us to detect (and prevent) the "output before input" case Johannes described above.